### ISSUE TITLE
Synthalingua seems to be "hearing voices" when streaming from m3u8/HLS

### ISSUE THREAD

**Issue by cgarvie** [cyberofficial/Synthalingua#84]
So I've installed Synthalingua on a vast.ai server with a dedicated GPU and am running the transcribe_audio.py and have tried both the --stream_translate and --stream_transcribe flags. 

On my laptop, I've run the attached python script to ingest audio from the microphone and turn it into an HLS stream available on localhost. Then I run ngrok infront of that to expose that to the web.

Then, simultaneously, on the server, I run:

$ python transcribe_audio.py --ram 12gb --stream_translate --stream_language English --stream_target_language French --stream_chunks 3 --energy_threshold 1000 --stream_original_text --stream https://camlang.ngrok.io/hls/stream.m3u8

The result is that when I speak, Synthalingua does indeed recognize, output, and translate the words into the target language. However, it is also constantly hearing other streams of audio that do not seem to exist. It commonly "hears" sentences spoken in Korean, English, and other languages (which it's not supposed to even be listening for), and it outputs them to the screen, and they are completely bizarre. For example:

-------------------------------------------------- en Original --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- Stream EN Translation --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- en Original --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- Stream EN Translation --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- en Original --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- Stream EN Translation --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- nn Original --------------------------------------------------
 H&M H&M H&M H&M H&M H&M H&M H&M H&M H&M H&M H&M H&M H&M
-------------------------------------------------- en Original --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- Stream EN Translation --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- nn Original --------------------------------------------------
 Hello everyone! Today I will show you how to make a delicious and delicious chicken breast with a very simple recipe . I will show you how to make this delicious
-------------------------------------------------- nn Original --------------------------------------------------

-------------------------------------------------- Stream EN Translation --------------------------------------------------
 Hello, I'm a newbie. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video. I'm going to make a new video.
-------------------------------------------------- nn Original --------------------------------------------------
 Hello everyone! Today I will show you how to make a delicious and delicious chicken breast with a simple recipe . I will show you how
-------------------------------------------------- en Original --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- Stream EN Translation --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- en Original --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- Stream EN Translation --------------------------------------------------
[Deleted correct microphone transcription of my speaking]
-------------------------------------------------- nn Original --------------------------------------------------

-------------------------------------------------- Stream EN Translation --------------------------------------------------
 Hello, I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl. I'm a little girl.
-------------------------------------------------- nn Original --------------------------------------------------
Hello everyone, welcome to my channel. Today I will show you how to make a simple and easy chocolate cake. Please watch the video until the end. First, make a chocolate cake. Add sugar and salt to the egg. Add the sugar and salt to the egg. Add the sugar and salt to the egg. Add the sugar and salt to the egg. Add the sugar and salt to the egg. Add the sugar and salt to the egg. Add the sugar and salt to the egg. Add the sugar and salt to the egg. Add the sugar and salt to the egg.
 
--

It's just completely random gibberish. It seems to come from tons of sources. It's not all recipes or youtube videos. 

Is this hallucination from the model? How do i regulate the microphone input to prevent it from hearing voices? I tried setting the 
--energy_threshold to 300, and to 1000, and changed the chunk size between 1 and 5 seconds.  This happens even if my microphone is totally silent.

Is there a Discord for this project btw?

Thanks

**Comment by cyberofficial** [comment]
This is a really good question, there shouldn't be ghosting as it's it only listens to the vocal range of the average male and average female of said languages , granted if sounds do come through that vocal range it would need to be white static or atmospheric interruption from the microphone or electrical interference such as charging a device and poorly placed microphone next to a powercord.

Are you near any high powered devices? is the microphone the laptop one or external? are you able to run something like RTX Voice for noise suppression? https://www.nvidia.com/en-us/geforce/guides/nvidia-rtx-voice-setup-guide/

**Comment by cyberofficial** [comment]
As for the discord, No. I haven't made one for it, haven't really gotten the thought of having one for it. probably could make one or something.

**Comment by cgarvie** [comment]
Rather than dig into the myriad possibilities, I've gotten approved for Youtube Livestreaming, and I'll just test that out as the stream source instead. I'll report back in a few days if that solved the issue.

**Comment by cyberofficial** [comment]
Any update to your issue?

**Comment by cgarvie** [comment]
I just tested it with Youtube Live as a stream source (but using same laptop mic, although in a totally different location) and had the same issue. This time, the model seems to hear "Thank you" repeated over and over as well as the occasional "welcome to my recipe channel, in this video...".

Cool project but I'm going to put my idea aside for now.

**Comment by cyberofficial** [comment]
I'm interested in helping you resolve this issue, this seems like an edge case that should not be ignored. If it happens to you (well is happening), then it could happen to others, and having a solution on hand would be great.

It doesn't have to be now, can be whenever, it's fine, if you simply don't want to go further then that's okay too, no pressure or problem.

Either way I'm aware now and can (well already) look into things and what may cause it.

**Comment by cyberofficial** [comment]
Moving to back burner, can be reopened later date if requested.

**Comment by cyberofficial** [comment]
> Hi Cyber, this seems to be the only project that has what I'm looking for I'm so glad you invested so much time onto this.
> [ .... ]
> Again, I thank you for putting your efforts into a project like this. I could not find anything else with a solution.


Hello! I have a giant refractor coming up to be released soon, https://github.com/cyberofficial/Synthalingua/tree/refactor

I'm currently working improving stuff and reliability of things. As of right now, I'm in my testing phase and am currently working methods to prevent ghosts from whispering into your audio.


AI models work by recognizing patterns - kind of like how you might try to lip read.
When lip reading, 'Hello there sir' might look like 'Yellow bear her' because the mouth movements are similar.

AI models do something similar but with much more complex patterns. In audio transcription, they're analyzing sound waves and looking for patterns they've learned represent certain words. But sometimes they 'see' patterns that aren't really there, or misinterpret background noise as speech.

The models used have been trained on thousands of examples, so it knows what certain word patterns typically 'look' like in audio data. But when the audio is unclear, has background noise, or the model encounters something slightly different from its training, it might confidently generate text that sounds reasonable but wasn't actually spoken.

In Short:
- It's pattern recognition gone wrong
- The AI doesn't "know" it's making something up - it genuinely "thinks" it heard something
- More training data usually helps, but doesn't eliminate the problem entirely
- It's particularly common with unclear audio, background noise, or edge cases the model wasn't well-trained on

 I can only reduce such things not fully prevent it. I am how ever working methods to clean up said noisy audio and trying to have ways of preventing some ghosting to occur and try to cancel out stuff. I've been working on the next update with many improvements and features, just the end game now is weeding out all the bugs that may have escaped testing.

**Comment by corshy1** [comment]
Thanks for the big update, Cyber.

I tried using it again but I don't think it has resolved my issue.

If there is no background noise. The translations are ok with only a bit of ghost transcriptions/translations.

If there is background noise, there's constant ghost audio. It's vomited out in lower models and higher ones adds to the queue every second almost non-stop.

I've tried maxxing the microphone threshold to see if I can make so it could only start recording when it detects speech but eventually it starts adding to the queue anyway and it builds up. It takes ages for the queue to clear.

I do wonder if perhaps I am not setting it properly in the GUI.

I have the following GUI settings:
Microphone mode
Isolate vocals ticked = DEFAULT
VRAM Size: 11gb-v2
Padded audio enabled (padded sound files 1)
CUDA enabled
Energy Threshold: 1000 enabled
Mic Chunk Size: 3
Repeat Protection: Ticked.
Stream Japanese.
Enable English Captions Ticked.

If you can, are there any recommended settings or logs I can provide you to help you analyze this?

**Comment by corshy1** [comment]
6GB model was definitely an improvement. It doesn't have to be instant. So long as it isn't too long and is reasonably accurate. I just follow what's translated on the CMD window.

I find the best outcome is enabling the following in microphone mode:

6GB model.
Phrase and Record timeout enabled but left to their defaults.
Enabled Padded Audio with Microphone Chunk Size (3 for long dialogue. 1 for short lines)
Using CUDA.
mdx_q with 14 cores/batch jobs.
Energy threshold enabled and set to 800. This is used as a noise gate/suppression. Preventing a lot of audio detecting streams that don't really need to be considered at all. Thus preventing a lot of the ghost transcripts. Though this did leave me with incidents where it's waiting for the next audio detection to process a batch if I had more than one chunk enabled. (A chunk/batch timeout feature upon no new audio stream detected could fix that)

Using this setup seemed to bring the most optimal results.

There is one problem with what I'm going for though.
Girls Frontline 2 sucks up all the GPU resources. When in the actual gameplay, processing audio translations grind to nearly a halt. So I'm forced to use the CPU for resources instead.
It still works out kind of well.

The only thing I noticed after a fair bit of testing is that it maybe harder for voices that are softer and have a lower pitch to be transcribed. Case in point: Nemesis (Japanese voice lines). Even with the Threshold Energy disabled. Her lines are sometimes not being picked up compared to others. I attached a video of the particular voiceline I am having trouble with.

[Nemesis Voiceline.zip](https://github.com/user-attachments/files/21326654/Nemesis.Voiceline.zip)
